tianshou/policy/base.py

```diff
@@ -60,6 +60,7 @@ def __init__(
         self.observation_space = observation_space
         self.action_space = action_space
         self.agent_id = 0
+        self.learning = False
```

Comment:
I don't like this new "state". IMHO, `training` state should be enough (with the former definition, corresponding to what you are calling `learning` in this PR). Even though this state actually exists, you always go from learning to testing. the new training state is therefore irrelevant in practice. For me, this clarification is just confusing me, by adding a new artificial state that is not used in practice.

Reply:
But if "learning" is not added, we need to add eps=0 in DQN update everywhere and exploration=False in DDPG/TD3/SAC everywhere, which possibly makes more bugs. Also with some layer like BatchNorm, it needs to distinguish the different state like [data collection in training (batchnorm.eval()), nn update (batchnorm.train()), data collection in testing (batchnorm.eval())], this is a real demand.

Reply:
I'll change the "training state" to "collecting state" to avoid confusion.

Reply:
Why would you collect samples without training and exploration ? For testing ?

Reply:
Yes

Reply:
So why is `training` not enough ? Is there any case where you do not want to collect data either ?

Reply:
I change the `learning` to `updating` and I think this time it is clearer:
```
During the training process, the policy has two main states: training state and testing state. The training state can be further divided into the collecting state and updating state.
```

Reply:
It's perfect! Look way more clear to me, good job !